processing/src/main/java/org/apache/carbondata/processing/loading/converter/impl/RowConverterImpl.java

```diff
@@ -130,9 +135,14 @@ public CarbonRow convert(CarbonRow row) throws CarbonDataLoadingException {
       }
       fieldConverters[i].convert(row, logHolder);
       if (!logHolder.isLogged() && logHolder.isBadRecordNotAdded()) {
-        badRecordLogger.addBadRecordsToBuilder(row.getRawData(), logHolder.getReason());
+        String reason = logHolder.getReason();
```

Comment:
If we had set the correct reason with row and column in `NonDictionaryFieldConverterImpl.convert()` instead of setting reason as `STRING_LENGTH_EXCEEDED_MESSAGE`, alone in it, we could have avoided changes in this file(RowConverterImpl) to fixup the the correct reason for `STRING_LENGTH_EXCEEDED_MESSAGE` again here at a common convert for all the columns of row.
That looks better. right ?

Reply:
We have to set the record also in log and we don't pass the rawdata to `NonDictionaryFieldConverterImpl.convert()` . So we have to do formatting in `RowConverterImpl`